### PR TITLE
(feat) Implement the deceased variant of the patient banner in the advanced search

### PR DESCRIPTION
**Pull request by jwnasambu**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

I implemented the next iteration of the patient banner when searching a deceased  patient on Advanced search  based on the design on this link https://zeroheight.com/23a080e38/p/6663f3-patient-header. i.e. Changed the colour of the deceased tag, background colour of the header and amended the font colours to match what's outlined in the design.

## Screenshots
Screenshot 2023-04-18 at 7 48 52 PM

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

File: packages/esm-patient-search-app/src/patient-search-page/patient-banner/banner/patient-banner.scss [hunk 1]
```diff
@@ -9,6 +10,36 @@
   grid-template-columns: 1fr auto;
 }
 
+.activePatientContainer {
+  background-color: $ui-02;
+}
+
+.deceasedPatientContainer {
+  background-color: colors.$gray-80;
```
Comment (ibacher) [inline packages/esm-patient-search-app/src/patient-search-page/patient-banner/banner/patient-banner.scss:18]: Maybe: `$color-gray-70` instead?

File: packages/esm-patient-search-app/src/patient-search-page/patient-banner/banner/patient-banner.scss [hunk 2]
```diff
@@ -9,6 +10,36 @@
...
+  .patientName {
+    color: $ui-01;
+  }
+
+  .demographics,
+  .identifiers {
+    color: $ui-02;
+  }
+
+  .actionsButtonText {
```
Comment (ibacher) [inline packages/esm-patient-search-app/src/patient-search-page/patient-banner/banner/patient-banner.scss:29]: Do we need to apply this color, i.e., is it any different from the standard color already there?

File: packages/esm-patient-search-app/src/patient-search-page/patient-banner/banner/patient-banner.scss [hunk 3]
```diff
@@ -9,6 +10,36 @@
...
+    color: colors.$blue-40;
+  }
+
+  &:focus,
+  &:hover {
+    background-color: colors.$gray-90;
```
Comment (ibacher) [inline packages/esm-patient-search-app/src/patient-search-page/patient-banner/banner/patient-banner.scss:35]: Maybe: $color-gray-70 instead?

File: packages/esm-patient-search-app/src/patient-search-page/patient-banner/banner/patient-banner.component.tsx [hunk 4]
```diff
@@ -110,12 +116,13 @@ const PatientBanner: React.FC<PatientBannerProps> = ({
...
           {!hideActionsOverflow && (
-            <div ref={overFlowMenuRef}>
+            <div className={styles.overflowMenuContainer} ref={overflowMenuRef}>
               <CustomOverflowMenuComponent
+                deceased={isDeceased}
                 menuTitle={
                   <>
                     <span className={styles.actionsButtonText}>{t('actions', 'Actions')}</span>{' '}
-                    <OverflowMenuVertical size={16} style={{ marginLeft: '0.5rem' }} />
+                    <OverflowMenuVertical size={16} style={{ marginLeft: '0.5rem', fill: '#78A9FF' }} />
```
Comment (ibacher) [inline packages/esm-patient-search-app/src/patient-search-page/patient-banner/banner/patient-banner.component.tsx:125]: Move all this to SCSS. Also, is it possible to use a standard color variable here?

File: packages/esm-patient-search-app/src/patient-search-page/patient-banner/ui-components/overflow-menu.component.tsx [hunk 5]
```diff
@@ -43,49 +45,25 @@ const CustomOverflowMenuComponent: React.FC<CustomOverflowMenuComponentProps> =
...
   return (
-    <div
-      data-overflow-menu
-      className="cds--overflow-menu"
-      style={{
-        width: 'auto',
-        height: 'auto',
-      }}
-      ref={wrapperRef}>
-      <Button
```
Comment (ibacher) [inline packages/esm-patient-search-app/src/patient-search-page/patient-banner/ui-components/overflow-menu.component.tsx:54]: I don't understand why this changes from a Carbon `button` to a standard `button` element?

File: packages/esm-patient-search-app/src/patient-search-page/patient-banner/ui-components/overflow-menu.component.tsx [hunk 6]
```diff
@@ -43,49 +45,25 @@ const CustomOverflowMenuComponent: React.FC<CustomOverflowMenuComponentProps> =
...
-        kind="ghost"
-        className={`cds--overflow-menu__trigger ${showMenu && 'cds--overflow-menu--open'}`}
+    <div data-overflow-menu className={`cds--overflow-menu ${styles.container}`} ref={wrapperRef}>
+      <button
+        className={`cds--btn cds--btn--ghost cds--overflow-menu__trigger ${showMenu && 'cds--overflow-menu--open'} ${
+          deceased && styles.deceased
```
Comment (ibacher) [inline packages/esm-patient-search-app/src/patient-search-page/patient-banner/ui-components/overflow-menu.component.tsx:51]:
```suggestion
          deceased ? styles.deceased : ''
```

**Review by ibacher — CHANGES_REQUESTED** [review]
Thanks @jwnasambu! This looks great! Just a few details to wrap up.

File: packages/esm-patient-search-app/src/patient-search-page/patient-banner/banner/patient-banner.component.tsx [hunk 7]
```diff
@@ -23,6 +23,7 @@ interface PatientBannerProps {
   onTransition?: () => void;
   hideActionsOverflow?: boolean;
   selectPatientAction: (evt: any, patientUuid: string) => void;
+  isActivePatient?: boolean;
```
[the remark this answers is not in the document]
Comment (jwnasambu) [reply packages/esm-patient-search-app/src/patient-search-page/patient-banner/banner/patient-banner.component.tsx:26]: @ibacher You are right! This is part of the changes for the next ticket. Thanks for the review!

File: packages/esm-patient-search-app/src/patient-search-page/patient-banner/banner/patient-banner.scss (hunk 2 above)
Comment (jwnasambu) [reply packages/esm-patient-search-app/src/patient-search-page/patient-banner/banner/patient-banner.scss:29]: We do need to apply that colour without it the identifier and demographic attributes are hardly seen.

**Comment by jwnasambu** [comment]
@ibacher Thanks for the review I have fixed the proposed changes.

File: packages/esm-patient-search-app/src/types/index.ts [hunk 8]
```diff
@@ -3,6 +3,7 @@ export interface SearchedPatient {
   uuid: string;
   identifiers: Array<Identifier>;
   person: {
+    [x: string]: any;
```
Comment (ibacher) [inline packages/esm-patient-search-app/src/types/index.ts:6]: It would be best to just add the attributes we need, especially as person only has so many attributes...

File: packages/esm-patient-search-app/src/patient-search-page/patient-banner/banner/patient-banner.scss (hunk 1 above)
Comment (denniskigen) [reply packages/esm-patient-search-app/src/patient-search-page/patient-banner/banner/patient-banner.scss:24]: Just realized that with my latest commit, I've basically undone this suggestion. The [design](https://zeroheight.com/23a080e38/p/6663f3-patient-header) specifies #393939 which [maps to](https://carbon-elements.netlify.app/colors/examples/preview/) `$colors.gray-80`.
